File: android/src/main/java/dev/luisbytes/capacitor/firebase/installations/FirebaseInstallationsPlugin.java
package dev.luisbytes.capacitor.firebase.installations;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import com.google.firebase.installations.FirebaseInstallations;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;

import androidx.annotation.NonNull;

@CapacitorPlugin(name = "FirebaseInstallations")
public class FirebaseInstallationsPlugin extends Plugin {

  @PluginMethod()
  public void getId(PluginCall call) {
    FirebaseInstallations.getInstance().getId()
      .addOnCompleteListener(new OnCompleteListener<String>() {
        @Override
        public void onComplete(@NonNull Task<String> task) {
          if (task.isSuccessful()) {
            JSObject ret = new JSObject();
            ret.put("id", task.getResult());
            call.resolve(ret);
          } else {
            call.reject("Id not found");
          }
        }
      });
  }

}

File: ios/Sources/FirebaseInstallationsPlugin/FirebaseInstallationsPlugin.swift
import Foundation
import Capacitor

import FirebaseInstallations

@objc(FirebaseInstallationsPlugin)
public class FirebaseInstallationsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "FirebaseInstallationsPlugin"
    public let jsName = "FirebaseInstallations"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getId", returnType: CAPPluginReturnPromise)
    ]

    @objc func getId(_ call: CAPPluginCall) {
        Task {
            do {
                let id = try await Installations.installations().installationID()
                call.resolve([
                    "id": id
                ])
            } catch {
                call.reject("Id not found")
            }
        }
    }
}
